container_registry/container_analysis/src/sample/samples_test.go

```diff
@@ -313,3 +258,166 @@ func TestPubSub(t *testing.T) {
 	sub.Delete(v.ctx)
 	teardown(t, v)
 }
+
+func TestPollDiscoveryOccurrenceFinished(t *testing.T) {
+	v := setup(t)
+
+	timeout := time.Duration(1) * time.Second
+	discOcc, err := pollDiscoveryOccurrenceFinished(v.imageUrl, v.projectID, timeout)
+	if err == nil || discOcc != nil {
+		t.Errorf("expected error when resourceUrl has no discovery occurrence")
+	}
+
+	// create discovery occurrence
+	noteId := "discovery-note-" + v.timestamp
+	noteReq := &grafeaspb.CreateNoteRequest{
+		Parent: fmt.Sprintf("projects/%s", v.projectID),
+		NoteId: noteId,
+		Note: &grafeaspb.Note{
+			Type: &grafeaspb.Note_Discovery{
+				Discovery: &discovery.Discovery{},
+			},
+		},
+	}
+	occReq := &grafeaspb.CreateOccurrenceRequest{
+		Parent: fmt.Sprintf("projects/%s", v.projectID),
+		Occurrence: &grafeaspb.Occurrence{
+			NoteName: fmt.Sprintf("projects/%s/notes/%s", v.projectID, noteId),
+			Resource: &grafeaspb.Resource{Uri: v.imageUrl},
+			Details: &grafeaspb.Occurrence_Discovered{
+				Discovered: &discovery.Details{
+					Discovered: &discovery.Discovered{
+						AnalysisStatus: discovery.Discovered_FINISHED_SUCCESS,
+					},
+				},
+			},
+		},
+	}
+	ctx := context.Background()
+	client, err := containeranalysis.NewGrafeasV1Beta1Client(ctx)
+	if err != nil {
+		t.Errorf("could not create client: %v", err)
+	}
+	defer client.Close()
+	_, err = client.CreateNote(ctx, noteReq)
+	created, err := client.CreateOccurrence(ctx, occReq)
+	if err != nil {
+		t.Errorf("createOccurrence(%s, %s): %v", v.imageUrl, v.noteID, err)
+	}
+
+	// poll again
+	timeout = time.Duration(20) * time.Second
```

Comment:
This is concerning since if the system happens to be slow, the test will fail. Please use `testutil.Retry` to try multiple times (like below).